src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -745,15 +746,20 @@ public UserProfileResponse getUserProfileFromUP(String idamId) {
 
 
         Optional<Object> resultResponse = validateAndGetResponseEntity(responseEntity);
-        if (resultResponse.isPresent() && resultResponse.get() instanceof UserProfileResponse profileResponse) {
-            if (nonNull(profileResponse.getIdamId())) {
-                return profileResponse;
-            }
-        }
-        return null;
+
+
+        return getProfileResponse(resultResponse);
     }
 
+    public UserProfileResponse getProfileResponse(Optional<Object> resultResponse) {
```

Comment:
its not best practice to use optional as method parameter

Reply:
As SonarQube is not providing warning or issue. Hence closing it